### ISSUE TITLE
Invalid key identifiers for Droplet creation

### ISSUE THREAD

**Issue by kofronpi** [Louisvh/TugShuttle#3]
Hello,

I installed all needed dependencies but the script fails on my machine (ubuntu 14.04, ruby 2.3.0, Tugboat 2.2.2) with this error message:

`Queueing creation of droplet 'tugshuttle-to-nyc1'...Failed to create Droplet: '1747168' are invalid key identifiers for Droplet creation.`

I succeed to create Droplets with Tugboat, this key ID and my `~/.tugboat` configuration file.

Maybe the tugboat key's id parsing is invalid because of an update on their side ?

Thanks !

**Comment by Louisvh** [comment]
Hey there,

TugShuttle grabs the key identifier using `keyID=$(cat ~/.tugboat | grep ssh_key: | sed -e "s/.*ssh_key://")` - somehow that's returning `XXXXX` for you. The id should be numeric, so that doesn't sound quite right. Could you check if the key id in `~/.tugboat` (ssh_key: _a_few_numbers_here_) matches the id of your actual key? Run `tugboat keys` to get a list of keys linked to your account with their respective id's. If that doesn't show any keys, you don't have an SSH key linked to your DO account - there's some info on how to fix that in the Prerequisites part of the Readme.

If your config file _does_ match an actual key identifier, I made a mistake in the script - please let me know if that's the case and I'll dig a little deeper to see what's causing it!

Kind regards,
Louis

**Comment by kofronpi** [comment]
Hi @Louisvh, 

I just obfuscated the key id by habit, I just edited the issue.

I am able to create a droplet with Tugboat, display its ip with a part of your bash script and then run sshuttle to it.

Here's the trace when I run your script:

```

kofronpi@home  λ  Tugshuttle (master) $ ./tugshuttle nyc1
Queueing creation of droplet 'tugshuttle-to-nyc1'...Failed to create Droplet: '1747168' are invalid key identifiers for Droplet creation.
Droplet fuzzy name provided. Finding droplet ID...error
Unable to find a droplet named 'tugshuttle-to-nyc1'.

IP of server: 

Attempting to add server to known host

Starting sshuttle proxy.
Binding: 12300
Listening on ('127.0.0.1', 12300).
[local sudo] Password: 

Keyboard interrupt: exiting.
```

And the output of `tugboat keys` shows the correct key and its id.

Thanks for your time !

**Comment by Louisvh** [comment]
Ah that makes more sense.

I just created an Ubuntu droplet to try and replicate your environment for a bit. I had no problems on 14.04 with Ruby 2.0 (2.3 wasn't in the repo but it's probably close enough) and Tugboat 2.2.2, so the only real variable of influence is the SSH key set-up.

Could you try running `tugboat create -r lon1 -s 512mb -i ubuntu-14-04-x64 -k 1747168 tugshuttle-to-lon1`? If that works but my script fails, my code is the offender, so that would be nice to isolate the problem.

While Googling around, I've seen some forum posts of people who had problems when having only a single SSH key linked to their account. I can't really test whether this is indeed a problem or not as I can't delete certain keys from my DO account right now. If you only have one SSH key linked, could you try linking a second (bogus) one?

If that's not it, could you post the output of `tugboat config`? (That basically outputs ~/.tugboat, but redacts your keys). There might be something funny going on in the rest of the config causing this issue.

**Comment by kofronpi** [comment]
I have several SSH keys linked to my DO account, and the command you provided works:

```
kofronpi@home  λ  Tugshuttle (master) $ tugboat create -r lon1 -s 512mb -i ubuntu-14-04-x64 -k 1747168 tugshuttle-to-lon1
Queueing creation of droplet 'tugshuttle-to-lon1'...Droplet created!
```

As for my config:

```
authentication:
  access_token:  [REDACTED]
ssh:
  ssh_user: root
  ssh_key_path: ~/.ssh/id_rsa_shuttle
  ssh_port: '22'
defaults:
  region: fra1
  image: ubuntu-14-04-x64
  size: 512mb
  ssh_key: '1747168'
  private_networking: 'false'
  backups_enabled: 'false'
  ip6: 'false'

```

**Comment by Louisvh** [comment]
Well this is a little embarrassing. Quotes around the ssh key id weren't being stripped. This isn't a problem if you've never used tugboat and let tugshuttle create ~/.tugboat (it doesn't add quotes to your key), but if there are quotes there, the script crashes on Droplet creation. This has been fixed in 0060b48. A simple git pull should fix the problem.

Thanks for reporting the bug though! I probably wouldn't have run into it myself, ever.
